### ISSUE TITLE
How to load image pairs?

### ISSUE THREAD

**Issue by RobinGerster** [open-mmlab/mmdetection#11494]
I would like to change the faster-rcnn pipeline. I want to do this by loading/processing my input images as normal. However, I also want to define a data augmentation (eg Bilateral Filter). Then in my training loop I would pass the original image forward as normal and simultaneously forward the augmented image (but this time with no gradient). The loss would be the average of my two.

Any idea how to implement something like this? I am especially confused about how I would deal with image pairs. 

One solution I though of would be to overwrite the "loss" function of the TwoStageDetector which has as input image batches (N,C,H,W). Then I could filter these and compute my loss again with no gradient. However, this seams inefficient/slow and also likely means that my images wont be processed correctly (as  they will no longer be mean/std adjusted).

If any expert here would be kind enough to share their insights into the correct approach that would be awesome!
